compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -1027,6 +1031,17 @@ abstract class PrepJSInterop extends plugins.PluginComponent
       ddef
     }
   }
+
+  private def checkInternalAnnotations(sym: Symbol): Unit = {
+    for {
+      ann <- compilerInternalAnnotations
+      annSym <- sym.getAnnotation(ann)
+      if annSym.pos.isDefined
```

Comment:
Yes, fail in `PreTyperComponent`. Because if the name is fully qualified, you are sure it actually refers to the one you want to exclude.